### ISSUE TITLE
NPE parsing referenced header parameters

### ISSUE THREAD

**Issue by pureJEagles** [swagger-api/swagger-parser#948]
When referencing a header parameter in my OAS spec, the parser throws NPE:

```
Exception while reading:
java.lang.NullPointerException: null
	at io.swagger.v3.parser.processors.ParameterProcessor.processParameters(ParameterProcessor.java:95)
	at io.swagger.v3.parser.processors.OperationProcessor.processOperation(OperationProcessor.java:39)
	at io.swagger.v3.parser.processors.PathsProcessor.processPaths(PathsProcessor.java:83)
	at io.swagger.v3.parser.OpenAPIResolver.resolve(OpenAPIResolver.java:49)
	at io.swagger.v3.parser.OpenAPIV3Parser.readLocation(OpenAPIV3Parser.java:53)
	at io.swagger.parser.OpenAPIParser.readLocation(OpenAPIParser.java:19)
	at org.openapitools.codegen.config.CodegenConfigurator.toClientOptInput(CodegenConfigurator.java:552)
	at org.openapitools.codegen.cmd.Generate.run(Generate.java:348)
	at org.openapitools.codegen.OpenAPIGenerator.main(OpenAPIGenerator.java:62)
Exception in thread "main" org.openapitools.codegen.SpecValidationException: There were issues with the specification. The option can be disabled via validateSpec (Maven/Gradle) or --skip-validate-spec (CLI).
 | Error count: 1, Warning count: 0
Errors: 
	-null

	at org.openapitools.codegen.config.CodegenConfigurator.toClientOptInput(CodegenConfigurator.java:569)
	at org.openapitools.codegen.cmd.Generate.run(Generate.java:348)
	at org.openapitools.codegen.OpenAPIGenerator.main(OpenAPIGenerator.java:62)
```

According to OAS spec, parser should be ignoring parameters with "in:header" and with name "Content-Type.  Seems like that same logic be applied when referencing header parameters too.  At the very least, there should be no NPE.

I'm seeing this problem using parser v.3.3.3.

Attaching a simple OAS spec in JSON format that can be used to reproduce the problem.
[headerError.txt](https://github.com/swagger-api/swagger-parser/files/2625568/headerError.txt)

**Comment by r-sreesaran** [comment]
@pureJEagles  Kindly attach the swagger file with issue was reproduced. If possible please raise a PR with failing test case. 

Thanks
R.Sreesaran

**Comment by Malligarjunan** [comment]
I am looking into it.

**Comment by Malligarjunan** [comment]
Looks like duplicate of https://github.com/swagger-api/swagger-parser/issues/934 #934

**Comment by pureJEagles** [comment]
Actually, it might be a duplicate of #684 but the fix doesn't seem to be on master or in the 3.3.3 release.

**Comment by gracekarina** [comment]
Thanks @Malligarjunan @pureJEagles Please let us know if the issue keeps happening.
Closed by #961
